### PR TITLE
disallowed-type: improve logic & add option in terraform/doc

### PR DESCRIPTION
**Pull request by xlr-8**
# Issue Type

- Feature Pull Request
- Bugfix Pull Request
- Documentation Pull Request

(All of the above)

## Summary

As reported [here](https://github.com/cristim/autospotting/issues/198) and [there](https://github.com/cristim/autospotting/pull/196#pullrequestreview-87979399). There were couple elements that could have been improved with the option.

This PR add/fixes:
* add terraform option & doc
* ensure no empty elements are within the list
* review logic, to give priority to ASG tags rather than global configuration

Closes: https://github.com/cristim/autospotting/issues/198

## Code contribution checklist

1. [x] The contribution fixes a single existing github issue, and it is linked
   to it.
1. [x] The code is as simple as possible, readable and follows the idiomatic Go
  [guidelines](https://golang.org/doc/effective_go.html).
1. [x] All new functionality is covered by automated test cases so the overall
  test coverage doesn't decrease.
1. [x] No issues are reported when running `make full-test`.
1. [x] Functionality not applicable to all users should be configurable.
1. [x] Configurations should be exposed through Lambda function environment
   variables which are also passed as parameters to the
   [CloudFormation](https://github.com/cristim/autospotting/blob/master/cloudformation/stacks/AutoSpotting/template.json)
   and
   [Terraform](https://github.com/cristim/autospotting/blob/master/terraform/autospotting.tf)
   stacks defined as infrastructure code.
1. [x] Global configurations set from the infrastructure stack level should also
   support per-group overrides using tags.
1. [x] Tags names and expected values should be similar to the other existing
   configurations.
1. [x] Both global and tag-based configuration mechanisms should be tested and
  proven to work using log output from various test runs.
1. [x] The logs should be kept as clean as possible (use log levels as
  appropriate) and formatted consistently to the existing log output.
1. [x] The documentation is updated to cover the new behavior, as well as the
   new configuration options for both stack parameters and tag overrides.
1. [x] A code reviewer reproduced the problem and can confirm the code
  contribution actually resolves it.

### CODE REVIEW

**Comment by xlr-8** [comment]
@CpuID & @binarylogic if you could give it a try with this updated version it would be kind of you

File: terraform/autospotting/variables.tf [hunk 1]
```diff
@@ -1,4 +1,14 @@
 # Autospotting configuration
+variable "autospotting_disallowed_instance_types" {
+  description = <<EOF
+Comma separated list of disallowed instance types for spot requests,
+in case you want to exclude specific types (also support globs).
+
+Example: 't2.*,m4.large'
```
Comment (cristim) [inline terraform/autospotting/variables.tf]: Are these supposed to be supported?

Please create some test cases for the globs as well.

**Comment by binarylogic** [comment]
Great! I'll have time next week to test the changes.

File: terraform/autospotting/variables.tf (hunk 1 above)
Comment (cristim) [reply terraform/autospotting/variables.tf]: No, I meant to ask if we do support glob matching for the black/white-listing logic. If so, we should have a few test cases for this.

Comment (cristim) [reply terraform/autospotting/variables.tf]: Never mind, I saw it now.

**Comment by xlr-8** [comment]
@binarylogic any update on this by chance?

**Comment by binarylogic** [comment]
Hi, unfortunately no, I haven't had time to test this yet. It'll probably be a couple of weeks before I can focus on that again, sorry 😞

**Comment by cristim** [comment]
I tested it, and it's pretty neat, thanks!

**Comment by cristim** [comment]
I noticed some naming and code inconsistencies between the way this feature was exposed to the group tags, adding a few more minor changes to that part.

**Comment by xlr-8** [comment]
Mind to develop? I haven't seen comments

File: cloudformation/stacks/AutoSpotting/parameters.yaml [hunk 2]
```diff
@@ -4,9 +4,9 @@
 
 # This file is handled automatically using the clouds-aws tool. It can be
 # installed using pip, or from https://github.com/elias5000/clouds-aws
-LambdaZipPath: nightly/lambda_build_458.zip
+LambdaZipPath: nightly/lambda_build_584.zip
 LambdaHandlerFunction: handler.Handle
 LambdaS3Bucket: cloudprowess
-LambdaMemorySize: '256'
+LambdaMemorySize: '1024'
```
Comment (xlr-8) [inline cloudformation/stacks/AutoSpotting/parameters.yaml]: You might as well change it in [terraform](https://github.com/cristim/autospotting/blob/master/terraform/variables.tf#L45) as well ;)

**Comment by binarylogic** [comment]
Agree, thank you for your work on this. I should be able to test next week as well.

**Comment by cristim** [comment]
Unfortunately this will require some stack updates of the tag used for per-group overrides, this will need to be announced once it lands.

File: cloudformation/stacks/AutoSpotting/parameters.yaml (hunk 2 above)
Comment (cristim) [reply cloudformation/stacks/AutoSpotting/parameters.yaml]: Done in  bcb5811

**Pull request merged**